core/runtime/src/main/java/io/quarkiverse/flow/providers/QuarkusTaskExecutorFactory.java

```diff
@@ -0,0 +1,11 @@
+package io.quarkiverse.flow.providers;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import io.quarkus.arc.DefaultBean;
+import io.serverlessworkflow.impl.executors.func.JavaTaskExecutorFactory;
+
+@DefaultBean
+@ApplicationScoped
+public class QuarkusTaskExecutorFactory extends JavaTaskExecutorFactory {
```

Comment:
Why do we need this? This executor factory will be implemented via Vertx most probably. Having a no-op class just for CDI it's not necessary since the SDK is already pulling it via ServiceLoader.

It's not working on native, is that it? If so, mind adding a comment to the class explaining why we need this no-op?

Reply:
> It's not working on native, is that it? If so, mind adding a comment to the class explaining why we need this no-op?

Yes @ricardozanini, it was necessary due to a native compilation error.

Error here:

> Caused by: java.util.ServiceConfigurationError: io.serverlessworkflow.impl.executors.TaskExecutorFactory: Provider io.serverlessworkflow.impl.executors.func.JavaTaskExecutorFactory not found

I added the JavaTaskExecutorFactory.class into the ServiceProviderBuildItem constructor.

Reply:
Hi @ricardozanini, I really do not know why those native tests are failling here, okay, I removed the no-op bean and now the pipeline are working.